src/Engine/System/CouplingSystem.hpp

```diff
@@ -0,0 +1,55 @@
+#ifndef RADIUMENGINE_COUPLING_SYSTEM_HPP
+#define RADIUMENGINE_COUPLING_SYSTEM_HPP
+
+#include <set>
+
+#include <Engine/System/System.hpp>
+
+namespace Ra {
+namespace Engine {
+
+/// Timed Systems are systems which can be disabled or paused, e.g. animation systems.
+class RA_ENGINE_API TimedSystem : public System {
+  public:
+    TimedSystem() {}
+    virtual ~TimedSystem() {}
+
+    /// Call this to enable / disable the system according to \p on.
+    virtual void play( bool on ) = 0;
+
+    /// Call this to enable the system for only one frame.
+    virtual void step() = 0;
+
+    /// Call this to reset the system, and its Components, to their first frame state.
+    virtual void reset() = 0;
+
+    /// Saves all the state data related to the \p frameID -th frame into a cache file.
+    virtual void cacheFrame( int frameID ) const = 0;
+
+    /// Restores the state data related to the \p frameID -th frame from the cache file.
+    /// \returns true if the frame has been successfully restored, false otherwise.
+    virtual bool restoreFrame( int frameID ) = 0;
+};
+
+/// Coupling Systems are responsible for transmitting calls to a bunch of other Systems.
+template <typename CSystem>

```

Comment:
Is `CSystem` always inheriting `CoupledSystem` ?
If yes, you don't need the template parameter.
If no, why do you need a template parameter, and why not using `System *` directly ?

Reply:
`CSystem` stands for `CoupledSystem` and which might provide the members required by the `CouplingSystem` managing it.
Having only `System*`, how can the `CouplingSystem` call the required members?

Reply:
So, just storing pointers of CoupledSystem should do the job, no ?